grant-project/grant-monorepo
# User Authentication UI (Pt 4 - Menus & Logout)

### What This Does

* Adds a menu to header auth to go to profile, settings, or sign out
* Adds a sign out and settings page (Settings is just a stub)
* Collapses menu at mobile to a hamburger menu
* Cleans up the auth state to be less jumpy while it's loading

### Steps to Test

* Sign in as a user, confirm that clicking the avatar on the top right opens a menu
* Sign out and refresh the page, confirm you stay signed out
* Reduce browser size, confirm you get a working hamburger menu

### Screenshots

<img width="280" alt="screen shot 2018-10-03 at 1 52 43 pm" src="https://user-images.githubusercontent.com/649992/46432330-aa48bd80-c713-11e8-8d54-793b44282f46.png">


<img width="999" alt="screen shot 2018-10-03 at 1 52 53 pm" src="https://user-images.githubusercontent.com/649992/46432338-ad43ae00-c713-11e8-9bb9-05784bdbeb89.png">


![menu](https://user-images.githubusercontent.com/649992/46432303-91400c80-c713-11e8-9a01-fd5aa24eb1ea.png)
